### PR TITLE
add initial pandas HDF fileformat

### PR DESCRIPTION
**Pull request by wkerzendorf**
The package was exactly what I'm looking for (for years I had some hack in TARDIS to do that). However, we store our reference files in the pandas HDF format (a specialized HDF5 format). I've added a simple initial implementation for the HDF format. If this is something that you would consider adding then I will clean it up and add tests

### CODE REVIEW

**Comment by wkerzendorf** [comment]
@astrofrog I added the code and the tests. I would add some more descriptions to the Readme. Is there anything else you would like for this PR?

**Review by astrofrog — APPROVED** [review]
Looks good, thanks! Before I merge can you add an entry to the changelog under section 0.4?

**Comment by wkerzendorf** [comment]
@astrofrog I'd like to rename the format `pd_hdf` if that is alright with you. In the future, we can just add `pd_excel` or other formats that people would like to save as.

**Comment by wkerzendorf** [comment]
@astrofrog ready to merge.

**Comment by wkerzendorf** [comment]
@astrofrog how do I add the dependencies now with tox enabled? I tried - but failed.

File: .travis.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ env:
         - TOXENV='test'
         - TOXARGS='-v'
         - TOXPOSARGS=''
+        - CONDA_DEPENDENCIES="pandas pytables"
```
Comment (astrofrog) [inline .travis.yml:27]: We don't use conda anymore for the CI so this should be removed

File: tests/test_pytest_arraydiff.py [hunk 2]
```diff
@@ -20,6 +20,10 @@ def test_succeeds_func_default():
 def test_succeeds_func_text():
     return np.arange(3 * 5).reshape((3, 5))
 
+@pytest.mark.array_compare(file_format='pd_hdf', reference_dir=reference_dir)
+def test_succeeds_func_pdhdf():
+    import pandas as pd
```
Comment (astrofrog) [inline tests/test_pytest_arraydiff.py:25]:
```
pd = pytest.importorskip('pytest')
```

That way the test will skip when pandas is not available.

**Review by astrofrog — CHANGES_REQUESTED** [review]
A couple of small requests below - add pandas and pytables under here:

https://github.com/astropy/pytest-arraydiff/blob/master/setup.cfg#L31

I'll open a separate PR to make those optional dependencies along with astropy.

File: .travis.yml (hunk 1 above)
Comment (wkerzendorf) [reply .travis.yml:27]: how do I install pandas?

**Comment by wkerzendorf** [comment]
@astrofrog okay - everything works - except macos because I need the HDF5 libraries. You guys don't use `conda` anymore I guess. How do I install these libraries now?

**Comment by pllim** [comment]
Try wheels? https://docs.h5py.org/en/stable/build.html#wheels

File: setup.cfg [hunk 3]
```diff
@@ -32,6 +32,8 @@ install_requires =
     pytest>=4.6
     numpy
     astropy
+    pandas
+    tables
```
Comment (wkerzendorf) [inline setup.cfg:37]:
```suggestion
    h5py
```

**Comment by wkerzendorf** [comment]
@astrofrog they all pass - can we get a merge please 😄

**Comment by wkerzendorf** [comment]
@astrofrog - is there anything missing for a merge?

**Comment by wkerzendorf** [comment]
@astrofrog I think I added everything - is there anything else?

**Comment by epassaro** [comment]
@wkerzendorf you can try removing my patch, maybe it's not necessary anymore.

File: CHANGES.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 0.6 (unreleased)
 ----------------
 
+- No changes yet.
```
Comment (astrofrog) [inline CHANGES.md:4]: Can you remove the 'no changes yet'?

File: CHANGES.md [hunk 5]
```diff
@@ -1,6 +1,9 @@
...
+- Add ability to compare to Pandas DataFrames and store them as HDF5 files
```
Comment (astrofrog) [inline CHANGES.md:5]: Add ``[#23]`` at the end here

**Review by astrofrog — CHANGES_REQUESTED** [review]
This looks good to me! Just a couple of minor requests below - also can you rebase to get rid of the merge commit? Thanks!

File: CHANGES.md (hunk 4 above)
Comment (wkerzendorf) [reply CHANGES.md:4]: so this was not put by me - but by @pllim

Comment (pllim) [reply CHANGES.md:4]: Feel free to remove it now that there is a change. :)

**Comment by astrofrog** [comment]
It looks like the changelog entry is now missing?

**Comment by wkerzendorf** [comment]
@astrofrog it seems only the travis and appveyor fail (potential no longer running). All Github actions pass.

**Comment by wkerzendorf** [comment]
@astrofrog the changelog entry is back. I think it should be good to go.

**Review by astrofrog — APPROVED** [review]
Looks good, thanks!

**Pull request merged**